docs/faq/ImmutableData.md

```diff
@@ -1,245 +1,145 @@
+- Immutable data management ultimately makes data handling safer.
+- Time-travel debugging requires that reducers be pure functions with no side effects, so that you can correctly jump between different states.
 
-Because you have to access your data via Immutable.JS’s own `get()` and `getIn()` methods, you can no longer use ES6’s destructuring and spread operators, making your code more verbose.
+In particular, Redux's use of shallow equality checking requires immutability if the store, and any connected components, are to be updated correctly. To see why, we need to understand the difference between shallow and deep equality checking in JavaScript.
 
-### Not suitable for small values that change often
+### How do shallow and deep equality checking differ?
+Shallow equality checking simply checks that two different _variables_ reference the same object; in contrast, deep equality checking must check every _value_ of two objects' properties.
 
-Immutable.JS works best for collections of data, and the larger the better. It can be slow when your data comprises lots of small, simple JavaScript objects, with each comprising a few keys of primitive values. Note, however, that this does not apply to the Redux state tree, which is represented as a large collection of data.
+A shallow equality check is therefore as simple (and as fast) as `a === b`, whereas a deep equality check involves a recursive traversal through the properties of two objects, comparing the value of each property at each step. 
 
-### Difficult to Debug
+It's for this improvement in performance that Redux uses shallow equality checking.
 
-Immutable.JS objects, such as `Map`, `List`, etc., can be difficult to debug, as inspecting such an object will reveal an entire nested hierarchy of Immutable.JS-specific properties that you don’t care about, while your actual data that you do care about is encapsulated several layers deep. 
+### How does Redux uses shallow equality checking?
+Redux shallowly compares the value returned from a reducer with the `state` parameter passed into it. If both objects are the same (i.e. the `state` parameter passed in references the same object as that returned by the reducer), then Redux assumes that the state has not been changed and so will leave the state tree unchanged.
 
-Fortunately, this can be easily overcome with the use of a browser extension such as the [Immutable.js Object Formatter](https://chrome.google.com/webstore/detail/immutablejs-object-format/hgldghadipiblonfkkicmgcbbijnpeog), which surfaces your data in Chrome Dev Tools, and hides Immutable.JS’s properties when inspecting your data.
+If, however, the returned value differs from the `state` object passed in (i.e. the `state` parameter passed in references a different object from that returned by the reducer), then Redux assumes a change has been made, and so will apply that change to the state tree.
```

Comment:
I reworded it...a lot! I closely inspected the Redux code and realised my assumptions of what was going on were wrong. So I've rewritten this bit substantially, which has a knock-on effect of the rest of the discussion on shallow checking and the issues it's caused.

It's added to the depth and length of the FAQ, but I think it's much clearer how and why shallow checking is used in Redux and React-Redux now.